src/api/customization.rs

```diff
@@ -80,6 +80,48 @@ pub trait Customization {
     /// Calling toggleAlwaysUv is preferred over enforcing alwaysUv here.
     fn enforce_always_uv(&self) -> bool;
 
+    /// Allows usage of enterprise attestation.
+    ///
+    /// # Invariant
+    ///
+    /// - Enterprise and batch attestation can not both be active.
+    /// - If the mode is VendorFacilitated, enterprise_attestation_mode() must be non-empty.
+    ///
+    /// For privacy reasons, it is disabled by default. You can choose between:
+    /// - EnterpriseAttestationMode::VendorFacilitated
+    /// - EnterpriseAttestationMode::PlatformManaged
+    ///
+    /// VendorFacilitated
+    /// Enterprise attestation is restricted to enterprise_attestation_mode(). Add your
+    /// enterprises domain, e.g. "example.com", to the list below.
+    ///
+    /// PlatformManaged
+    /// All relying parties can request an enterprise attestation. The authenticator
+    /// trusts the platform to filter requests.
+    ///
+    /// To enable the feature, send the subcommand enableEnterpriseAttestation in
+    /// AuthenticatorConfig. An enterprise might want to customize the type of
+    /// attestation that is used. OpenSK defaults to batch attestation. Configuring
+    /// individual certificates then makes authenticators identifiable.
+    ///
+    /// OpenSK prevents activating batch and enterprise attestation together. The
+    /// current implementation uses the same key material at the moment, and these
+    /// two modes have conflicting privacy guarantees.
+    /// If you implement your own enterprise attestation mechanism, and you want
+    /// batch attestation at the same time, proceed carefully and remove the
+    /// assertion.
+    fn enterprise_attestation_mode(&self) -> Option<EnterpriseAttestationMode>;
+
+    /// Lists relying party IDs that can perform enterprise attestation.
+    ///
+    /// # Invariant
+    ///
+    /// - If the mode is VendorFacilitated, enterprise_attestation_mode() must be non-empty.
+    ///
+    /// This list is only considered if the enterprise attestation mode is
+    /// VendorFacilitated.
+    fn enterprise_rp_id_list(&self) -> Vec<String>;
```

Comment:
@ia0 unlike the previous one, this one does have overhead because the caller only needs to check whether the list contains a value. Should we do the backing store trick to expose a &[&str] interface? Or ignoring is fine?

Reply:
Usage might always change, right? So if we don't optimize before, the same argument should still apply?

Reply:
I think we should expose:

```rust
fn is_enterprise_rp_id(&self, rp_id: &str) -> bool;
```

If we need something else in the future, we will assess at that point. For the other `Vec<String>` we actually needed to iterate/merge its content.

Reply:
Done, but PTAL

Reply:
I understood the comment as "only expose `is_enterprise_rp_id`, remove `enterprise_rp_id_list`", but let's check with @ia0 to clarify.

Reply:
Yes indeed, but I realize now that we need to check whether the list is empty in tests. So we still need either `enterprise_rp_id_list` or `is_enterprise_rp_id_list_empty` (or similar).

I guess the simplest is to add `#[cfg(feature = "std")]` to `enterprise_rp_id_list`.

Reply:
Yeah that's why I kept the enterprise_rp_id_list API. Now I changed it to only exposed in std.